cita-executor/core/src/tests/amend_data_test.rs

```diff
@@ -66,53 +108,38 @@ fn call_vm(
 #[test]
 fn call_amend_data() {
     let mut state = get_temp_state();
-    let super_admin_address: Address = "0000000000000000000000000000000000012345".into();
-    state.super_admin_account = Some(super_admin_address.clone());
     let key = H256::from(42);
     let value = H256::from(42);
     let storage_address: Address = "0000000000000000000000000000000000055555".into();
     let mut data: Bytes = storage_address.to_vec();
     data.append(&mut key.to_vec());
     data.append(&mut value.to_vec());
-    // non admin sender
-    let sender = Address::default();
-    let address: Address = reserved_addresses::AMEND_ADDRESS.into();
-    let mut params = ActionParams::default();
-    params.address = address;
-    params.code_address = address;
-    params.sender = sender.clone();
-    params.origin = sender.clone();
-    params.gas = U256::from(10000);
-    params.value = ActionValue::Apparent(3.into());
-    params.data = Some(data.clone());
-    let result = call_vm(&mut state, params);
+    // Sender is not super admin
+    // `value=3` means the operation of amending kv
+    let result = transact(&mut state, &data, U256::from(3), false);
     assert!(result.is_err());
 
-    // sender is admin
-    let mut params = ActionParams::default();
-    let sender: Address = super_admin_address.clone();
-    params.address = address;
-    params.code_address = address;
-    params.sender = sender.clone();
-    params.origin = sender.clone();
-    params.gas = U256::from(10000);
-    params.value = ActionValue::Apparent(3.into());
-    params.data = Some(data.clone());
-    let result = call_vm(&mut state, params);
+    // Sender is super admin
+    let result = transact(&mut state, &data, U256::from(3), true);
     assert!(result.is_ok());
 
-    // get value from key
     let mut data: Bytes = storage_address.to_vec();
     data.append(&mut key.to_vec());
+    // Get value from key use transact interface
+    // `value=4` means the operation of getting value from key.
+    let result = transact(&mut state, &data, U256::from(4), true);
+    assert!(result.is_ok());
+    // TODO Add output of transact in executive.rs when amend get-value
+    // println!("result: {:?}", result.clone().unwrap());
+
+    // Get value from key use call interface
+    let address: Address = reserved_addresses::AMEND_ADDRESS.into();
```

Comment:
It's better to slipt this test into two cases `superAdmin` or `Not superAdmin`

Reply:
It just fix the older test. so if follow the old style.

There are othere problems beside what you say:

* the other amend operation
* the output(check the todo), maybe it's the problem of progress of executive.